CRM/Contribute/BAO/Contribution.php

```diff
@@ -910,6 +910,26 @@ protected static function recordPaymentActivity($contributionId, $participantId,
     self::addActivityForPayment($entityObj, $financialTrxn, $activityType, $component, $contributionId);
   }
 
+  /**
+   * Get memberships realted to the contribution.
+   *
+   * @param int $contributionID
+   *
+   * @return array
+   */
+  protected static function getRelatedMemberships($contributionID) {
+    $contribution = new CRM_Contribute_BAO_Contribution();
+    $contribution->id = $contributionID;
```

Comment:
this 'double loading' of contribution is because we expect to stop passing the contribution object later